rest_server/SiodbLoggerPool.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2019-2020 Siodb GmbH. All rights reserved.
+// Use of this source code is governed by a license that can be found
+// in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	LogFileSizeMin              uint32 = 1 * 1024 * 1024
+	LogFileExpirationTimeoutMin uint64 = 1 * 60
+)
+
+var (
+	FATAL_INIT_ERROR             = 2
+	FATAL_CANNOT_CREATE_LOG_FILE = 3
+)
+
+type SiodbLoggerPool struct {
+	siodbLogger []*SiodbLogger
+}
+
+func (loggerPool *SiodbLoggerPool) ConfigGinLogger() {
+	gin.SetMode(gin.ReleaseMode)
+	gin.DisableConsoleColor()
+	gin.DefaultWriter = ioutil.Discard
+}
+
+func (loggerPool *SiodbLoggerPool) LogRequest(c *gin.Context, latency time.Duration) {
+	path := c.Request.URL.Path
+	raw := c.Request.URL.RawQuery
+	if raw != "" {
+		path = path + "?" + raw
+	}
+
+	loggerPool.Output(INFO, "Request | %3d | %13v | %15s | %15v | %s | %-7s | %s",
+		c.Writer.Status(),
+		latency,
+		c.ClientIP(),
+		c.Request.UserAgent(),
+		c.Request.Method,
+		c.Request.Proto,
+		path,
+	)
+}
+
+func (loggerPool *SiodbLoggerPool) Output(logLevel int, s string, v ...interface{}) error {
+	log := FormattedOutput(logLevel, s, v...)
+	for _, siodbLogger := range loggerPool.siodbLogger {
+		siodbLogger.Output(logLevel, log)
+	}
+	return nil
+}
+
+func (loggerPool *SiodbLoggerPool) Trace(s string, v ...interface{}) {
+	loggerPool.Output(TRACE, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Debug(s string, v ...interface{}) {
+	loggerPool.Output(DEBUG, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Info(s string, v ...interface{}) {
+	loggerPool.Output(INFO, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Warning(s string, v ...interface{}) {
+	loggerPool.Output(WARNING, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Error(s string, v ...interface{}) {
+	loggerPool.Output(ERROR, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) FatalAndExit(code int, s string, v ...interface{}) {
+	loggerPool.Output(FATAL, s, v...)
```

Comment:
how will this work if error code is "can't create log file" when writing to the same channel where this error occurred? Old file is already closed and new failed to open. This should be worked out somehow. Easiest I think is to check that `logger.out` is nil, do not output to it, just return, and set it to nil after file close.